src/main/java/seedu/address/model/lesson/Lesson.java

```diff
@@ -0,0 +1,74 @@
+package seedu.address.model.lesson;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+import seedu.address.model.assignment.Deadline;
+import seedu.address.model.assignment.ModuleCode;
+import seedu.address.model.assignment.Name;
+import seedu.address.model.assignment.Task;
+
+public class Lesson extends Task {
+    /**
+     * Every field must be present and not null.
+     */
+    public Lesson(Name name, Deadline deadline, ModuleCode moduleCode) {
+        super(name, deadline, moduleCode);
+        requireAllNonNull(name, deadline, moduleCode);
+    }
+
+    public Deadline getDeadline() {
+        return super.getTime();
+    }
+
```

Comment:
Naming the deadline field time may be more appropriate for the Lesson class as it doesn't really make sense for a Lesson to have a deadline.

Reply:
Amended!